packages/expo-notifications/ios/EXNotifications/Notifications/NotificationCenterManager.swift

```diff
@@ -0,0 +1,131 @@
+import ExpoModulesCore
+import Foundation
+
+/**
+ Protocols that NotificationCenterManager delegates may implement
+ */
+public protocol NotificationPresentationDelegate: AnyObject {
+  func willPresent(_ notification: UNNotification, completionHandler: @escaping (UNNotificationPresentationOptions) -> Void)
+}
+
+public protocol NotificationResponseDelegate: AnyObject {
+  func didReceive(_ response: UNNotificationResponse, completionHandler: @escaping () -> Void)
+}
+public protocol NotificationSettingsDelegate: AnyObject {
+  func openSettings(_ notification: UNNotification?)
+}
+
+public protocol NotificationRegistrationSuccessDelegate: AnyObject {
+  func didRegister(_ deviceToken: String)
+}
+
+public protocol NotificationRegistrationFailureDelegate: AnyObject {
+  func didFailRegistration(_ error: Error)
+}
+
+/**
+ Singleton that sets itself as the UserNotificationCenter delegate,
+ and calls its own delegates in response to notification center calls.
+ */
+@objc(EXNotificationCenterManager)
+public class NotificationCenterManager: NSObject,
+  UNUserNotificationCenterDelegate,
+  NotificationRegistrationFailureDelegate,
+  NotificationRegistrationSuccessDelegate {
+  @objc
+  public static let shared = NotificationCenterManager()
+
+  var delegates: [AnyObject] = []
```

Comment:
If we have just one protocol for delegates (see a comment above), I guess we won't need an array of delegates.

Reply:
There will be multiple modules registering as delegates, so I think we will still need the array